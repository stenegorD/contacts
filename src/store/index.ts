import { combineReducers, configureStore } from "@reduxjs/toolkit";
import { contactsApi } from "./reducers/contacts/api";


const rootReducer = combineReducers({
    // contacts: contactsSlice.reducer,
    [contactsApi.reducerPath]: contactsApi.reducer,
});



export const store = configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware().concat(
            contactsApi.middleware,
        )
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
