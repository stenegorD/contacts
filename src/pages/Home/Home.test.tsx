import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';
import { Home } from '.';
import { store } from '../../store';


describe('Home', () => {
    it('should render Home component correctly', () => {
        const { container } = render(
            <Provider store={store}>
                <MemoryRouter>
                    <Home />
                </MemoryRouter>
            </Provider>
        );

        expect(container).toMatchSnapshot();
    });
});