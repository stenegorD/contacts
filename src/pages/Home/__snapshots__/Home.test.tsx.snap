// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > should render Home component correctly 1`] = `
<div>
  <div
    class="flex justify-center gap-8 m-10 flex-col md:flex-row"
  >
    <div
      class="flex flex-col gap-2 w-full h-full sticky top-[40px] bottom md:w-contact-form"
    >
      <h2
        class="text-xl"
      >
        Create Contact
      </h2>
      <form
        autocomplete="off"
        class="flex flex-col gap-2"
        novalidate=""
      >
        <div
          class="field"
        >
          <label
            class="label"
            for="firstName"
          >
            Name
          </label>
          <input
            class="input"
            id="firstName"
            name="firstName"
            placeholder="Name"
          />
          <p
            class="error"
          />
        </div>
        <div
          class="field"
        >
          <label
            class="label"
            for="lastName"
          >
            Surname
          </label>
          <input
            class="input"
            id="lastName"
            name="lastName"
            placeholder="Surname"
          />
          <p
            class="error"
          />
        </div>
        <div
          class="field"
        >
          <label
            class="label"
            for="email"
          >
            Email
          </label>
          <input
            class="input"
            id="email"
            name="email"
            placeholder="Email"
            type="email"
          />
          <p
            class="error"
          />
        </div>
        <button
          class="submit-button"
          type="submit"
        >
          Add contact
        </button>
      </form>
    </div>
    <div
      class="flex flex-col gap-2"
    >
      <h2
        class="text-xl"
      >
        Contact List
      </h2>
      <div
        class="w-full md:w-contact-list"
      >
        Loading...
      </div>
    </div>
  </div>
</div>
`;
